docs/Submitting_jobs.rst

```diff
@@ -0,0 +1,411 @@
+          depend on it were submitted, the submission process would fail.
+
+
+.. _submitting_jobs_usage_gc3pie:
+
+Submitting jobs to a ``GC3Pie`` backend
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When using the ``GC3Pie`` backend, EasyBuild will create separate tasks for each installation to be performed and
+supply them to GC3Pie, which will then take over and pass the installations through as jobs to the available
+resource(s) (see also :ref:`submitting_jobs_cfg_job_backend_config`).
+
+To ensure that the installations are performed in the order dictated by the software dependency graph, dependencies
+between installations are specified to GC3Pie as inter-task dependencies. GC3Pie will then gradually feed the
+installations to its available resources as their dependencies have been satisfied.
+
+Any log messages produced by GC3Pie are included in the EasyBuild log file, and are tagged with ``gc3pie``.
+
+See also :ref:`submitting_jobs_examples_gc3pie_backend`.
+
+.. note:: The ``eb`` process will not exit until the full set of tasks that GC3Pie was provided with has been processed.
+          An overall progress report will be printed regularly (see also :ref:`submitting_jobs_cfg_job_polling_interval`).
+          As such, it is advised to run the ``eb`` process in a screen/tmux session when using the GC3Pie backend for
+          ``--job``.
+
+
+.. _submitting_jobs_examples:
+
+Examples
+--------
+
+.. _submitting_jobs_examples_gc3pie_cfg:
+
+Example configurations for GC3Pie job backend
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When using GC3Pie as a job backend, a configuration file must be provided via ``--job-backend-config``.
+This section includes a couple of examples of GC3Pie configuration files (see also
+https://gc3pie.readthedocs.org/en/latest/users/configuration.html).
+
+Example GC3Pie configuraton for local system
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code:: ini
+
+  [resource/localhost]
+  enabled = yes
+  type = shellcmd
+  frontend = localhost
+  transport = local
+  max_cores_per_job = 1
+  max_memory_per_core = 10GiB
+  max_walltime = 100 hours
+  # this doubles as "maximum concurrent jobs"
+  max_cores = 4
+  architecture = x86_64
+  auth = none
+  override = no
+  resourcedir = /tmp/gc3pie
+
+
+Example GC3Pie configuration for PBS/TORQUE
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code:: ini
+
+  # this is only needed if connecting through SSH to the cluster
+  # frontend; otherwise you can remove this [auth/*] section, and just
+  # use 'auth=none' in the resource definition
+  [auth/myuser]
+  type = ssh
+  username = me
+  
+  [resource/pbs]
+  enabled = yes
+  type = pbs
+
+  # use settings below when running GC3Pie on the cluster front-end node
+  frontend = localhost
+  transport = local
+  auth = none
+  # use settings below when connecting through SSH to the cluster
```

Comment:
By the way: would the connect-over-SSH work at all with EB?  I understand EB would read/search the `.eb` and module files on the host where it runs, so you would need them in the same exact location as on the cluster frontend?